infra/azure/k8s/cilium/kube-system.yaml

```diff
@@ -0,0 +1,221 @@
+apiVersion: "cilium.io/v2"
+kind: CiliumNetworkPolicy
+metadata:
+  name: default-deny
+  namespace: kube-system
+spec:
+  endpointSelector: {}
+  ingress:
+    - {}
+  egress:
+    - {}
+---
+apiVersion: "cilium.io/v2"
+kind: CiliumNetworkPolicy
+metadata:
+  name: kube-dns
+  namespace: kube-system
+spec:
+  endpointSelector:
+    matchLabels:
+      k8s-app: kube-dns
+  ingress:
+    # Allow kube-dns to receive DNS requests from any endpoint in the cluster
+    # Policies for those endpoints will determine if and how they can communicate with kube-dns
+    - fromEntities:
+        - cluster
+      toPorts:
+        - ports:
+          - port: "53"
+            protocol: ANY
+    # Allow konnectivity-agent to communicate with kube-dns
+    - fromEndpoints:
+        - matchLabels:
+            app: konnectivity-agent
+            k8s:io.kubernetes.pod.namespace: kube-system
+      toPorts:
+        - ports:
+          - port: "9153"
+            protocol: ANY
+  egress:
+    # Allow kube-dns to send DNS requests in the cluster
+    - toEntities:
+        - cluster
+      toPorts:
+        - ports:
+          - port: "53"
+            protocol: ANY
+    # Allow kube-dns to communicate with the Kubernetes API server
+    - toEntities:
+        - kube-apiserver
+      toPorts:
+        - ports:
+          - port: "443"
+            protocol: TCP
+    # Allow kube-dns to communicate with the Azure DNS server
+    - toCIDR:
+      - 168.63.129.16/32
+      toPorts:
+        - ports:
+          - port: "53"
+            protocol: ANY
+---
+apiVersion: "cilium.io/v2"
+kind: CiliumNetworkPolicy
+metadata:
+  name: hubble-ui
+  namespace: kube-system
```

Comment:
The hubble rules are going to be the same for all K8s implementations, I think it would be sensible to put those in a different file and provision them seperately. That way, it will be easier to manage the other rules in `kube-system` which may need to be changed for different platforms.